app/src/main/java/de/danoeh/antennapod/dialog/IntraFeedSortDialog.java

```diff
@@ -31,7 +31,7 @@ public void openDialog() {
 
         int idxCurrentSort = -1;
         for  (int i = 0; i < values.length; i++) {
-            if (currentSortOrder == values[i]) {
+            if (currentSortOrder == values[i] || currentSortOrder == null) {
```

Comment:
Thanks for looking into this.

I think it looks a bit strange to have this check inside the for loop. Wouldn't it also work to initialize `idxCurrentSort` with 0 instead?

Reply:
I tried to figure out the logic in why someone sets it to -1 but couldnt see a reason. Rather than create an issue later, I went with a null check. Nothing unusual was observed if we just start at 0 and it selects the default sort. Pushing the fix.